### DISCUSSION TITLE
Redundancy and physical distribution of open-meteo commercial servers?

### DISCUSSION THREAD

**Discussion by rkhamilton** [open-meteo/open-meteo#652] [General]
We have recently launched a commercial weather app that serves both forecast and historical weather data for locations globally. We have been unhappy with the reliability of our existing weather data provider and open-meteo's features looks like a good fit for us.

Can one of the project maintainers comment on the suitability of the physical open-meteo systems for commercial use? I have seen a photo on the open meteo blog that look like consumer desktop hardware. Is this project being hosted in multiple data centers with redundant hardware and failover support? Is it hosted in a single site? A startup founder's garage with a single server that must periodically be taken down for maintenance? We are trying to understand the business risk of using open-meteo and the uptime that our customers would experience. The project's [status page](https://status.open-meteo.com) is not showing any data as I write this. What has been the historical uptime?

Thank you!

**Comment by patrick-zippenfenig** [comment]
Hi, thanks for asking. Consumer grade hardware you saw in some of the blog posts is only used converting historical weather data. All real-time weather processing takes place on servers in different datacenters. But even those tasks are now mostly running on dedicated servers.

APIs offers to customers as part of the commercial API subscription use servers in Europe and North America. Due to load-balancing and Geo-based latency routing, this provides good reliability and latency for most users. With the uptake of more and more users in Asia, additional servers in Singapore will be deployed in the next weeks. The free API is currently not using Geo-based latency routing, due to cost (AWS Route56 would be more expensive than running servers for the free API). This might change at some point with different geo-based latency providers.

Weather data processing chain is fairly fault-tolerant against single provider failures. E.g. the next US government shutdown won't effect Open-Meteo significantly (Note: Government shutdowns do not take down critical GRIB weather forecasting services). Data processing runs on multiple servers as well to protect from hardware failures.

The Open-Meteo infrastructure is fairly flat and simple to scale. There is no central database that needs a high-availability cluster or regular maintenance to resolve performance issues. Weather data is stored in plain files that can be distributed to servers directly (which is also distributed through an [AWS S3 Bucket](https://github.com/open-meteo/open-data)). This results in a resilient mesh infrastructure and each individual API node can operate without external dependencies.

The only limit to quickly scale up more and more servers is the total amount of data. Storage (especially fast NVMe SSD storage) and bandwidth are the main constraints. For an optimal cost-benefit ratio, servers from [Hetzner](https://www.hetzner.com), [OVH](https://www.ovhcloud.com/en/) and [Contabo](https://contabo.com/en/) are used. No services use residential ISPs ;-)

The status page has not been updated for quite some time. I will add all relevant API endpoints for commercial customers soon. I wanted to switch to the [Atlassian status page service](https://www.atlassian.com/software/statuspage), but other items have higher priority.

**Reply by rkhamilton** [reply]
Thanks! We'll start doing some prototyping with your free API.
